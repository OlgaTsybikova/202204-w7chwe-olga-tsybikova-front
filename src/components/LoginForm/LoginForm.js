import LoginFormStyled from "./LoginFormStyled";

const LoginForm = () => {
  return (
    <LoginFormStyled autoComplete="off" noValidate>
      <label htmlFor="username">
        Username:
        <input id="username" type="text" />
      </label>
      <label htmlFor="password">
        Password:
        <input id="password" type="text" />
      </label>
      <button type="submit">LOG IN</button>
    </LoginFormStyled>
  );
};
export default LoginForm;
