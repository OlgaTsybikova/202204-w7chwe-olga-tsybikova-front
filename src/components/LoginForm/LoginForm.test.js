import LoginForm from "./LoginForm";
import "@testing-library/jest-dom";
import reactTestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../redux/store/store";

describe("Given a LoginForm component", () => {
  describe("When it is calledOn", () => {
    test("Should identify if the spy function was called or not", () => {
      const spyFunction = jest.fn();
      spyFunction();
      expect(spyFunction).toHaveBeenCalled();
    });
  });
  test("Then it should render correctly when there are no items", () => {
    const tree = reactTestRenderer
      .create(
        <Provider store={store}>
          <LoginForm />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
