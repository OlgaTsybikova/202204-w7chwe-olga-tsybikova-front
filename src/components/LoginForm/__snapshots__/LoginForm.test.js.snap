// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginForm component Then it should render correctly when there are no items 1`] = `
<form
  autoComplete="off"
  className="sc-bczRLJ hDLlWZ"
  noValidate={true}
  onSubmit={[Function]}
>
  <label
    htmlFor="username"
  >
    Username:
    <input
      id="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="password"
  >
    Password:
    <input
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </label>
  <button
    disabled={true}
    type="submit"
  >
    LOG IN
  </button>
</form>
`;
